eclair-core/src/main/scala/fr/acinq/eclair/swap/SwapResponses.scala

```diff
@@ -0,0 +1,80 @@
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.swap
+
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Satoshi}
+import fr.acinq.eclair.payment.Bolt11Invoice
+import fr.acinq.eclair.wire.protocol.{HasSwapId, OpeningTxBroadcasted, SwapInAgreement, SwapInRequest}
+
+object SwapResponses {
+
+  sealed trait Response {
+    def swapId: String
+  }
+
+  sealed trait Success extends Response
+
+  sealed trait Fail extends Response
+
+  sealed trait Error extends Fail
+
+  case class SwapOpened(swapId: String) extends Success {
+    override def toString: String = s"swap $swapId opened successfully."
+  }
+
+  case class UserCanceled(swapId: String) extends Fail {
+    override def toString: String = s"swap $swapId canceled by user."
+  }
+
+  case class PeerCanceled(swapId: String) extends Fail {
+    override def toString: String = s"swap $swapId canceled by peer."
+  }
+
+  case class InvalidMessage(swapId: String, behavior: String, message: HasSwapId) extends Fail {
+    override def toString: String = s"swap $swapId canceled due to invalid message during $behavior: $message."
+  }
+
+  case class LocalError(swapId: String, t: Throwable) extends Error {
+    override def toString: String = s"swap $swapId local error: $t."
+  }
+
+  case class SwapError(swapId: String, reason: String) extends Error {
+    override def toString: String = s"swap $swapId swap error: $reason."
+  }
+
+  case class InsufficientBalanceForReceive(swapId: String, amount: Satoshi, availableForReceive: Satoshi) extends Error {
+    override def toString: String = s"swap $swapId error: requested amount of $amount sat > available channel balance to receive of $availableForReceive sat."
+  }
+
+  case class InsufficientBalanceForSend(swapId: String, amount: Satoshi, availableForSend: Satoshi) extends Error {
+    override def toString: String = s"swap $swapId error: requested amount of $amount sat > available channel balance to send of $availableForSend sat."
+  }
+
+  case class InsufficientOnChainBalance(swapId: String, amount: Satoshi, maxPremium: Satoshi, onChainBalance: Satoshi) extends Error {
+    override def toString: String = s"swap $swapId error: requested amount of $amount + $maxPremium maximum premium > confirmed on-chain balance of $onChainBalance."
+  }
+
+  case class InternalError(swapId: String, reason: String) extends Error {
+    override def toString: String = s"swap $swapId internal error: $reason."
+  }
+
+  sealed trait Status extends Response
+  case class SwapInStatus(swapId: String, actor: String, behavior: String, channelId: ByteVector32, request: SwapInRequest, agreement_opt: Option[SwapInAgreement] = None, invoice_opt: Option[Bolt11Invoice] = None, openingTxBroadcasted_opt: Option[OpeningTxBroadcasted] = None) extends Status {
+    override def toString: String = s"$actor[$behavior]: $swapId, $channelId, $request, $agreement_opt, $invoice_opt, $openingTxBroadcasted_opt"
+  }
```

Comment:
I don't think we should expose that much internal information, `actor` and `behavior` are really internal details IMO.
But I think it would be valuable to have something like:

```scala
sealed trait SwapInStatus extends Status
case class AwaitingSwapInAgreement(...) extends SwapInStatus
case class AwaitOpeningTxConfirmation(...) extends SwapInStatus
...
```

The list of statuses we should expose should map to the _protocol_ statuses where we're waiting for an external event (a response from our peer or a blockchain event like tx confirmation). We should _not_ include transient statuses (such as `CreatingOpeningTx` for example), because these statuses will automatically move to a "real" status once an internal call completes, they would only clutter the status trait for no good reason. When the actor is in one of these transient statuses and receives a `GetStatus` request, it should simply delay it and it will eventually be handled once we're in a "real" status.

This is quite subtle, does it make sense (and do you think it's a good idea)?

Reply:
Good point! I've updated GetStatus to emit the minimal set of user meaningful messages, and moved the verbose information to the debug log: [3403272](https://github.com/ACINQ/eclair/pull/2496/commits/3403272cb76c79b3aa00f5739c9b07692c760123)